dist/README.md

```diff
@@ -1,37 +1,102 @@
 
 ```sh
-./bin/generate-secure-secrets
+USERNAME=grist \
+PASSWORD=grist \
+DEFAULT_EMAIL=gristadmin@example.com \
+GRIST_DOMAIN=grist.example.com \
+./bin/bootstrap-environment
 ```
 
-<!-- TODO: Logically separate out the crypto part of setup from the
-other bits required, such as APP_HOME_URL or persist location-->
-This will use Authelia to generate some cryptographic information needed in `config/secrets`, as well as an `.env` file in the current directory with some defaults for running the Compose image. You may wish to inspect the `.env` file to change some defaults, but leave the cryptographic choices alone. Then run as usual,
+This will use Authelia and OpenSSL to generate some cryptographic
+information needed in `persist/secrets`, as well as an `.env` file in
+the current directory with some defaults for running the Compose
+image.
+
+You may wish to inspect the `.env` file to change some defaults,
+
+```sh
+nano .env
+```
+
+but leave the cryptographic choices alone.
+
+Then run as usual,
 
 ```sh
 docker compose up
 ```
 
 This creates the following default setup:
 
-* Grist at http://grist.localhost
-<!--TODO: let user configure the default username/password -->
-* Default username: `test`
-* Default password: `test`
+* Grist at https://$GRIST_DOMAIN
+* Username for the admin user: $USERNAME
+* Password for the admin user: $PASSWORD
+* Email for the admin user: $DEFAULT_EMAIL
+
+If you want to add, modify or delete users, you can do this in
+`persist/users_database.yml`. All of the instructions needed
+to do that are in the file.
+
+Compose will start, in this order, Traefik, Authelia, Dex, and finally
+Grist.
+
+# Using systemd
+
+If everything looks fine, you may interrupt Compose (hit Ctrl-C), run
+`docker compose down` and use systemd to handle the Grist Docker process:
+
+```sh
+sudo systemd start grist
+```
+
+The Compose logs will then be available in systemd's journal service,
+for example,
+
+```sh
+sudo journalctl -xeu grist
+```
+
+This will ensure that Grist restarts cleanly if you need to stop or
+restart the server it's running on.
+
+# Configuration
+
+The following environment variables can be configured:
 
-If you want to add, modify or delete users, you can do this in `config/authelia/users_database.yml`. All of the instructions needed to do that are in the file.
+* `USERNAME`
+* `PASSWORD`
+* `DEFAULT_EMAIL`
+* `GRIST_DOMAIN`
+* `TEAM`
+* `TRAEFIK_ENABLE_DASHBOARD`
+* `HTTPS_METHOD`
+* `GOOGLE_CLIENT_ID`
+* `GOOGLE_CLIENT_SECRET`
+* `MICROSOFT_CLIENT_ID`
+* `MICROSOFT_CLIENT_SECRET`
+* `PERSIST_DIR`
+* `SECRETS_DIR`
 
-Compose will start, in this order, Traefik, Authelia, Dex, and finally Grist, at the URL specified by `APP_HOME_URL` (by default, at `http://grist.localhost` as above).
+They are documented in the generated `.env` file. They may be changed,
+but some require re-generating files. For example, changing
+`GRIST_DOMAIN` may require re-generating a self-signed certificate.
```

Comment:
Can we add something like:

> This can be done by re-running the bootstrap script with new parameters, although this will reset any variables you've changed manually.

Reply:
I don't currently have a super-clean way of doing this, and I'm not sure how to give concise instructions in this case. Basically, if you change the domain name, you need to also delete the cert in order to generate a new self-signed cert, but that's just one example. I am kind of hoping that people try to re-run the bootstrap script, the script tells them what it refused to overwrite and why, and if they want to overwrite something they need to delete the file so it gets re-generated instead.

Reply:
Maybe we just identify the common cases (such as domain name changes) and provide explicit instructions for those?

> * GRIST_DOMAIN - If GRIST_DOMAIN is changed and you are using the automatically created self-signed certificates, you will need to delete your existing certificate and re-run the bootstrap script to create a new certificate.

Or maybe an FAQ / Common issues section later?